lib/bracket-matcher.coffee

```diff
@@ -24,6 +24,7 @@ class BracketMatcher
     "‹": "›"
 
   toggleQuotes: (includeSmartQuotes) ->
+    return if @pairedCharacters? and includeSmartQuotes is (@pairedCharacters isnt @defaultPairs)
```

Comment:
For full scope support, this line ensures that we don't update `pairedCharacters` in every `toggleQuotes` call (only when the config value and/or scope changes).